cluster.go

```diff
@@ -1741,3 +1741,76 @@ func diffPeers(peers1, peers2 []peer.ID) (added, removed []peer.ID) {
 	}
 	return
 }
+
+// RepoGC performs garbage collection sweep on all peers' IPFS repo.
+func (c *Cluster) RepoGC(ctx context.Context) (*api.GlobalRepoGC, error) {
+	_, span := trace.StartSpan(ctx, "cluster/RepoGC")
+	defer span.End()
+	ctx = trace.NewContext(c.ctx, span)
+
+	members, err := c.consensus.Peers(ctx)
+	if err != nil {
+		logger.Error(err)
+		return nil, err
+	}
+	lenMembers := len(members)
+
+	ctxs, cancels := rpcutil.CtxsWithCancel(ctx, lenMembers)
+	defer rpcutil.MultiCancel(cancels)
+
+	repoGCsResp := make([]*api.RepoGC, lenMembers, lenMembers)
+
+	errs := c.rpcClient.MultiCall(
+		ctxs,
+		members,
+		"Cluster",
+		"RepoGCLocal",
+		struct{}{},
+		rpcutil.CopyRepoGCSliceToIfaces(repoGCsResp),
+	)
+
+	// clubbing repo gc responses of all peers into one
+	globalRepoGC := api.GlobalRepoGC{PeerMap: make(map[string]*api.RepoGC)}
+	for i, resp := range repoGCsResp {
+		e := errs[i]
+
+		if e == nil {
+			globalRepoGC.PeerMap[peer.IDB58Encode(resp.Peer)] = resp
+			continue
+		}
+
+		if rpc.IsAuthorizationError(e) {
+			logger.Debug("rpc auth error:", e)
+			continue
+		}
+
+		logger.Errorf("%s: error in broadcast response from %s: %s ", c.id, members[i], e)
+
+		resp.Error = e.Error()
+		globalRepoGC.PeerMap[peer.IDB58Encode(resp.Peer)] = resp
+	}
+
+	return &globalRepoGC, nil
+}
+
+// RepoGCLocal perform garbage collection only on the local IPFS deamon.
+func (c *Cluster) RepoGCLocal(ctx context.Context) (*api.RepoGC, error) {
+	_, span := trace.StartSpan(ctx, "cluster/RepoGCLocal")
+	defer span.End()
+	ctx = trace.NewContext(c.ctx, span)
+
+	repoGC := &api.RepoGC{
```

Comment:
This is overly convoluted. 

```go
resp, err := c.ipfs.RepoGC(ctx)
if err != nil {
  return nil, err
}
resp.Peer = c.id
resp.Peername = c.config.Peername
return resp, nil
```

Reply:
can't `return nil, err`
finding an error doesn't mean response is nil https://github.com/ipfs/ipfs-cluster/pull/777/files#diff-66308423c5638970cde2dd02574201c3R847

When cluster's `RepoGC` calls `RepoGCLocal`, it will fill RepoGC struct with peer and error. I have tried to produce same response, even if `RepoGCLocal` rpc call is called directly.

Reply:
I re-iterate my ask to do this this way. If you had an error, forget about the response. It's an error. This makes assumptions about how something is implemented somewhere else when it does not need to.

When RepoGC calls RepoGCLocal, if there's an error, the response will be discarded anyways (see rpc_api.go).